src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/ReplicationCheckerTest.java

```diff
@@ -0,0 +1,158 @@
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskOperation;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskOperationType;
+import uk.gov.hmcts.reform.wataskmanagementapi.db.TCExtendedContainerDatabaseDriver;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskHistoryResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.entity.TaskResource;
+import uk.gov.hmcts.reform.wataskmanagementapi.repository.TaskResourceRepository;
+import uk.gov.hmcts.reform.wataskmanagementapi.services.operation.ReplicationChecker;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.sql.DataSource;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.awaitility.Awaitility.await;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.wataskmanagementapi.cft.enums.CFTTaskState.UNASSIGNED;
+
+@ActiveProfiles("replica")
+public class ReplicationCheckerTest extends SpringBootIntegrationBaseTest {
+
+    @Autowired
+    TaskResourceRepository taskResourceRepository;
+
+    @Autowired
+    DataSource dataSource;
+
+    @Autowired
+    TaskHistoryResourceRepository taskHistoryResourceRepository;
+    @Autowired
+    ReportableTaskRepository reportableTaskRepository;
+    @Autowired
+    TaskAssignmentsRepository taskAssignmentsRepository;
+
+    SubscriptionCreator subscriptionCreator;
+
+    @Autowired
+    TCExtendedContainerDatabaseDriver tcDriver;
+
+    @Autowired
+    ReplicationChecker replicationChecker;
+
+    @Value("${spring.datasource.jdbcUrl}")
+    private String primaryJdbcUrl;
+
+
+    @Value("${spring.datasource-replica.jdbcUrl}")
+    private String replicaJdbcUrl;
+
+    CFTTaskDatabaseService cftTaskDatabaseService;
+    MIReportingService miReportingService;
+
+    JdbcDatabaseContainer container;
+    JdbcDatabaseContainer containerReplica;
+
+    @BeforeEach
+    void setUp() {
+        subscriptionCreator = new SubscriptionCreator("repl_user", "repl_password",
+            "wa_user", "wa_password");
+        miReportingService = new MIReportingService(taskHistoryResourceRepository, taskResourceRepository,
+            reportableTaskRepository,
+            taskAssignmentsRepository,
+            subscriptionCreator);
+        CFTTaskMapper cftTaskMapper = new CFTTaskMapper(new ObjectMapper());
+        cftTaskDatabaseService = new CFTTaskDatabaseService(taskResourceRepository, cftTaskMapper);
+
+        container = TCExtendedContainerDatabaseDriver.getContainer(primaryJdbcUrl);
+        containerReplica = TCExtendedContainerDatabaseDriver.getContainer(replicaJdbcUrl);
+        Testcontainers.exposeHostPorts(container.getFirstMappedPort(), containerReplica.getFirstMappedPort());
```

Comment:
This will mean an inconsistent test because the creation of the subscription is done in a different thread.  The test should wait until the subscription is setup correctly before executing any scenarios.  
See `master` here: https://github.com/hmcts/wa-task-management-api/blob/master/src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/services/MIReplicaReportingServiceTest.java#L90

Reply:
Done